### PR TITLE
chore: Remove cached dep + transitive ones

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
So many dependencies (esp out of date ones) makes me sad while waiting on compliation, though it gives me time to putz while doing so

# Changes
Implement `cached` ourself and remove some duplicate dependencies (if you expand out `Cargo.lock` you will see quite a few transitive dependencies removed

### CODE REVIEW

File: server/src/lib.rs [hunk 1]
```diff
@@ -952,19 +951,48 @@ impl ConnectionManager for ConnectionManagerImpl {
...
         .map_err(|e| Box::new(e) as _)
         .context(RemoteServerConnectError)?;
     let client = write::Client::new(connection);
-    Ok(Return::new(Arc::new(RemoteServerImpl { client })))
+    let cc = CachedConnection {
+        value: Arc::new(RemoteServerImpl { client }),
+        was_cached: false,
+    };
+
+    // reacquire lock (which might result in thundering herd problems)
```
Comment (carols10cents) [inline server/src/lib.rs:991]: 🐃 🐄 🏇🏻 🦒 🦏

**Review by carols10cents — APPROVED** [review]
LGTM!

**Comment by mkmik** [comment]
I'm particularly frustrated by our slow builds, so I should cheer at this one.

However, I'm working right now at the storage api router and used `cached` there too. I realized I wanted entries to expire after 60s, so I just added `time = 60` to the annotation. I'm not sure I want to re-implement that manually.

**Comment by mkmik** [comment]
perhaps I could find a smaller caching crate with less deps?

File: server/src/lib.rs (hunk 1 above)
Comment (mkmik) [reply server/src/lib.rs:991]: fwiw `cached` is also subject to this
```
The function-cache is not locked for the duration of the function's execution, so initial (on an empty cache) concurrent calls of long-running functions with the same arguments will each execute fully and each overwrite the memoized value as they complete.
```

**Comment by alamb** [comment]
> perhaps I could find a smaller caching crate with less deps?

If you can, sure. Given you are about to use more features of the `caching` crate keeping it around makes sense to me. I had a few minutes to kill this morning between drop off and the next kid activity and figured I could remove deps for a small amount of code.

I will try some other ways to reduce the dependency tree

**Comment by mkmik** [comment]
I'll find a smaller lib (or write one). This is indeed has an obscene amount., for what it does.

Also, I don't like the fact that it's based on a macro. Yes, I get it, transparently memoization is cool etc; but it doesn't work on methods, etc...

(I'd like a cargo plugin that yells at you if you add a dep with a lot of transitive deps. The more transitive deps the louder it should yell at you)

**Pull request closed without merging**